pennylane/transforms/qcut.py

```diff
@@ -827,6 +824,118 @@ def expand_fragment_tapes_mc(
+    """
+    Helper function to reshape results and find out degrees of communication
+    graph
+    """
+    results = [qml.math.flatten(r) for r in results]
+    results = np.array(results, dtype=object)
+    results = results.reshape((len(communication_graph), shots)).T
+
+    return results
+
+
+def qcut_processing_fn_sample(
+    results: Sequence, communication_graph: MultiDiGraph, shots: int
+) -> List:
+    """
+    Function to postprocess samples for the :func:`cut_circuit_mc() <pennylane.cut_circuit_mc>`
+    transform. This removes superfluous mid-circuit measurement samples from fragment
+    circuit outputs.
+
+    .. note::
+
+        This function is designed for use as part of the sampling-based circuit cutting workflow.
+        Check out the :func:`qml.cut_circuit_mc() <pennylane.cut_circuit_mc>` transform for more details.
+
+    Args:
+        results (Sequence): a collection of execution results generated from the
+            random expansion of circuit fragments over measurement and preparation node configurations
+        communication_graph (nx.MultiDiGraph): the communication graph determining connectivity
+            between circuit fragments
+        shots (int): the number of shots
+
+    Returns:
+        List[tensor_like]: the sampled output for all terminal measurements over the number of shots given
+    """
+    res0 = results[0]
+    results = _reshape_results(results, communication_graph, shots)
+    out_degrees = [d for _, d in communication_graph.out_degree]
+
+    samples = []
+    for result in results:
+        sample = []
+        for fragment_result, out_degree in zip(result, out_degrees):
+            sample.append(fragment_result[: -out_degree or None])
+        samples.append(np.hstack(sample))
+    return [qml.math.convert_like(np.array(samples), res0)]
+
+
+def qcut_processing_fn_mc(
+    results: Sequence,
+    communication_graph: MultiDiGraph,
+    settings: np.ndarray,
+    shots: int,
+    classical_processing_fn: callable,
+):
+    """
+    Function to postprocess samples for the :func:`cut_circuit_mc() <pennylane.cut_circuit_mc>`
+    transform. This takes a user-specified classical function to act on bitstrings and
+    generates an expectation value.
+
+    .. note::
+
+        This function is designed for use as part of the sampling-based circuit cutting workflow.
+        Check out the :func:`qml.cut_circuit_mc() <pennylane.cut_circuit_mc>` transform for more details.
+
+    Args:
+        results (Sequence): a collection of execution results generated from the
+            random expansion of circuit fragments over measurement and preparation node configurations
+        communication_graph (nx.MultiDiGraph): the communication graph determining connectivity
+            between circuit fragments
+        settings (tensor_like): each element is one of 8 unique values that determines and tracks the specific
```

Comment:
```suggestion
        settings (np.ndarray): Each element is one of 8 unique values that tracks the specific
```
We have control over the type of `settings`.